_source/_posts/2022-01-04-angular-auth0-quickly.md

```diff
@@ -0,0 +1,177 @@
+
+```shell
+npx @angular/cli@13 new auth0-demo --routing
+```
+
+Select your favorite stylesheet format. The selection you make doesn't matter for this example.
+
+## Add Auth with OpenID Connect
+
+To add authentication with Auth0, you'll first need a [free Auth0 account](https://auth0.com/signup). Install the [Auth0 CLI](https://github.com/auth0/auth0-cli#installation) and run `auth0 login` to register your account. Then, run `auth0 apps create`. Specify a name and description of your choosing. Select **Single Page Web Application** and use `http://localhost:4200/home` for the Callback URL. Specify `http://localhost:4200` for the rest of the URLs.
+
+{% img blog/angular-auth0/auth0-apps-create.png alt:"auth0 apps create" width:"800" %}{: .center-image }
+
+Once you have a new Angular app and Auth0 OIDC set up, you can use [OktaDev Schematics](https://github.com/oktadev/schematics#angular--auth0) to add OAuth 2.0 and OIDC support to your Angular app.
+
+```shell
+ng add @oktadev/schematics --auth0
+```
+
+You'll be prompted for an issuer and client ID. You should have these from the OIDC app you just created. The issuer will be displayed right above your client ID.
+
+```shell
+=== dev-0ua1y-go.us.auth0.com application created
+
+  CLIENT ID            TJgZxGnlSkqUe6JfMFSipZcFsbpl6LS2
+```
+
+This process will perform the following steps for you:
+
+1. Install the [Auth0 Angular SDK](https://github.com/auth0/auth0-angular).
+2. Add `src/app/auth-routing.module.ts` with your OIDC configuration and initialization logic.
+3. Configure an `AuthHttpInterceptor` that adds an `Authorization` header with an access token to outbound requests.
+4. Create a `HomeComponent` and configure it with authentication logic.
+5. Update unit tests for `AppComponent` and `HomeComponent` to mock Auth0.
+
+To see all the changes it makes to your files, see [this pull request on GitHub](https://github.com/oktadev/auth0-angular-example/pull/1/files). 
+
+### Use the Auth0 Console instead of the CLI
+
+You can also use the Auth0 Console to create an OIDC app:
+
+* [Log in](https://auth0.com/auth/login) to Auth0 or [create an account](https://auth0.com/signup) if you don't have one. Go to **Applications** > **Create Application**.
+* Choose **Single Page Web Applications** as the application type and click **Create**.
+* Click **Angular**, then the **Settings** tab.
+* Add `http://localhost:4200/home` as an Allowed Callback URL and `http://localhost:4200` as a Logout URL.
+* Specify `http://localhost:4200` as an Allowed Origin and click **Save Changes** at the bottom.
+
+### Test your Angular authentication flow
+
+Run `ng serve` in your app, and you should see a login button at `http://localhost:4200/home`.
+
+{% img blog/angular-auth0/auth0-login-button.png alt:"Auth0 Login button" width:"800" %}{: .center-image }
+
+Click the **Login** button and sign in with one of the users that's configured in your Auth0 application, or sign up as a new user.
```

Comment:
```suggestion
Click the **Login** button and sign in with one of the configured users for your Auth0 application. Because this is just a Demo, you probably haven't done that - more likely, you should sign up as a new user.
```